bots/psuteam7bot/movement.js

```diff
@@ -0,0 +1,409 @@
+            return candidateB;
+    }while(!(dirA.x === direction.x && dirA.y === direction.y));
+
+    return location;
+}
+
+/*More complex pathfinding algorithm to check passable path of a robot's current location/ position towards a destination and get the closest possible position to destination
+*
+*/
+/*
+movement.pathFinding = (self, destination) => {
+    return;
+}
+movement.pathfinding = (self, destination) => {
+    const maxDist = SPECS.UNITS[self.me.unit].SPEED;
+    let distance = getDistance(self.me, destination);
+    const maxFuelCost = (distance * SPECS.UNITS[self.me.unit].FUEL_PER_MOVE);
+
+    //Looking through 'API questions' discord channel, 'karbonite' and 'fuel' seems to be the way to get global team's karbonite and fuel
+    if(fuel < maxFuelCost)
+        distance = Math.floor(fuel/FUEL_PER_MOVE);
+
+    let {dx, dy} = destination;
+    let direction = getRelativeDirection(self.me, destination);
+    let x = -1;
+    let y = -1;
+
+    //Get number of XY tile moves to get to the location in moverange that is closest to destination
+    if(distance > maxDist)
+    {
+        let {distX, distY} = getDistanceXY(self.me, destination);
+        let rDist = Math.sqrt(maxDist);
+
+        if(distX > rDist && distY > rDist)
+        {
+            dx = rDist;
+            dy = rDist;
+        } 
+        else if(distX < rDist && distY > rDist)
+        {
+            dy = maxDist-dx;
+        }
+        else if(distX > rDist && distY < rDist)
+        {
+            dx = maxDist-dy;
+        }
+
+        x = dx * direction.x;
+        y = dy * direction.y;
+    }
+    x += self.me.x;
+    y += self.me.y;
+
+    //Goal is the location in moverange that is closest to destination, may be impassable
+    let goal = {x,y};
+}
+*/
+
+/*A simple movement function for robot movement from point A to Point B,
+*Input: self        -   The robot unit
+*       destination -   The destination 'position/ location' object {x, y}, assumes passable
+*Output:    retVal  -   A 'position/ location' object {x, y} within moverange of robot location, which is passable and closest to destination
+*                   -   OR {-1, -1} if distance from robot location to destination <= 0
+*/
+movement.moveTowards = (self, destination) => {
+    const maxDist = SPECS.UNITS[self.me.unit].SPEED;
+    let fullMap = self.map;
+    let robotMap = self.getVisibleRobotMap();
+    let distance = movement.getDistance(self.me, destination);
+    const maxFuelCost = (distance * SPECS.UNITS[self.me.unit].FUEL_PER_MOVE);
+
+    //Looking through 'API questions' discord channel, 'karbonite' and 'fuel' seems to be the way to get global team's karbonite and fuel
+    if(fuel < maxFuelCost)
+        maxDist = Math.floor(fuel/FUEL_PER_MOVE);
+
+
+    //Case 0: No movement
+    if(distance <= 0)
+        return {x: -1, y: -1};
+
+    //Case 1: move towards dest
+    let current = {
+        x: self.me.x, 
+        y: self.me.y
+    };
+    let previous = self.me.previous;
```

Comment:
Should probably update `robot.js` to include `this.previous` in the constructor since you access it here.